### PR TITLE
feat(combo-box): introduce combo box

### LINKED ISSUE #44: Component: Combo box
Probably start with some basic features.

### PR DESCRIPTION
**Pull request by asudoh**
Fixes #44.

### CODE REVIEW

File: src/components/dropdown/dropdown.ts [hunk 1]
```diff
@@ -506,15 +548,14 @@ class BXDropdown extends HostListenerMixin(FocusMixin(LitElement)) {
         <div
           role="button"
```
Comment (cal-smith) [inline src/components/dropdown/dropdown.ts]: Just want to verify that `role="button"` is valid if the trigger isn't focussable ...

Comment (asudoh) [reply src/components/dropdown/dropdown.ts]: Good point! Fixed.

**Comment by asudoh** [comment]
Thank you for reviewing and finding the issues @cal-smith!

> - If I'm typing and press space it auto selects the first item

Couldn't reproduce it with a space key. Reproduced it with hitting backspace key to make the `<input>` empty and I guess this is what you meant, but please don't hesitate to correct me if I'm wrong here.

> - when I arrow key up and down the cursor moves to the start and end of the text, as well as moving the selection ... I'm not sure what's correct here, but that feels wrong

Makes sense, fixed.

> - when I clear the filter (with backspace) it doesn't always clear the selection right away (make a selection -> close the menu -> open the menu -> clear selection -> `x` still visible until tab out)

Great catch! Fixed.

**Comment by cal-smith** [comment]
> Couldn't reproduce it with a space key. Reproduced it with hitting backspace key to make the <input> empty and I guess this is what you meant, but please don't hesitate to correct me if I'm wrong here.

I'm still seeing it ... the easiest way to replicate seems to be to start typing an option title, then press space, like so:

giphy

**Comment by asudoh** [comment]
Got it - Thanks alot @cal-smith for clarifying! Fixed.

**Pull request merged**